tests/functional/models/test_device.py

```diff
@@ -374,6 +374,17 @@ def test_get_supervisor_state(self):
             self.balena.models.device.get_supervisor_state(device['uuid'])
         self.assertIn('No online device(s) found', cm.exception.message)
 
+    def test_get_supervisor_target_state(self):
+        device = self.helper.create_device()[1]
+
+        # should be rejected if the device doesn't exist.
+        with self.assertRaises(Exception) as cm:
+            self.balena.models.device.get_supervisor_target_state('9999999')
+        self.assertIn('No online device(s) found', cm.exception.message)
+        # should be rejected if the device doesn't exist.
```

Comment:
@xginn8, I think this case is about offline device, not the one doesn't exist. Also I'm able to GET the state of offline device so we shouldn't use `assertRaises` here